### PR TITLE
Make hash facade more typesafe

### PR DESCRIPTION
**Pull request by eed3si9n**

### CODE REVIEW

File: core/src/main/scala/sjsonnew/Facade.scala [hunk 1]
```diff
@@ -46,6 +46,31 @@ trait Facade[J] {
...
+ */
+trait FContext[J] {
+  def addField(s: String): Unit
+  def add(v: J): Unit
+  def finish: J
+  def isObj: Boolean
+}
+
+/**
+ * Facade is a type class that describes how JSON AST elements of
```
Comment (dwijnand) [inline core/src/main/scala/sjsonnew/Facade.scala]: ExtractorFacade

File: core/src/main/scala/sjsonnew/SimpleFacade.scala [hunk 2]
```diff
@@ -30,7 +30,7 @@ import scala.collection.mutable
  * Facade[J] also uses FContext[J] instances, so implementors will
```
Comment (dwijnand) [inline core/src/main/scala/sjsonnew/SimpleFacade.scala]: SimpleBuilderFacade

Comment (dwijnand) [inline core/src/main/scala/sjsonnew/SimpleFacade.scala]: and above as well

File: core/src/main/scala/sjsonnew/Facade.scala (hunk 1 above)
Comment (dwijnand) [inline core/src/main/scala/sjsonnew/Facade.scala]: and below as well

Comment (eed3si9n) [inline core/src/main/scala/sjsonnew/Facade.scala]: Fixed.

**Pull request merged**